source/libNsdl/src/sn_grs.c

```diff
@@ -23,7 +23,7 @@
  */
 #include <string.h>
 #include <stdlib.h>
-
+#include <assert.h>
```

Comment:
I thought that the NDEBUG should get rid of any extra crap that assert() brings and that define should be anyway set on the release build versions.

That tr_assert() is a really good idea. Currently mbed-os' assert() is mostly visible only on debugger. If the assert fail or any other mbed os exception happens on tests, the test framework does not get any output and logs just "Dut1 timed out" or something equally useful.